### PR TITLE
Add xpub derivation paths to PSBT inputs

### LINKED ISSUE #54: Should we set the keypath in PSBTs inputs
That'd be YA thing to think about but maybe some HW require it..

### PR DESCRIPTION
**Pull request by darosior**
This is an alternative to #96 which modifies the public key type used in derived descriptors to also comport information about the key source instead of taking a mapping directly through the public API. This is IMO much less of a breakage and much cleaner as less footguny.
This however adds an assumption on the derived descriptors.

Happy to bikeshed!

Fixes #54
Closes #96

### CODE REVIEW

**Comment by edouardparis** [comment]
Seems cool, the new DerivedPublicKey is clean
untestedACK b4afbd2051289464655ea4a2de3eab462460e822

File: src/scripts.rs [hunk 1]
```diff
@@ -38,6 +44,109 @@ use serde::de;
...
+        }
+
+        if s.chars().next().expect("Size checked above") != '[' {
+            return Err(ScriptCreationError::DerivedKeyParsing);
+        }
+
+        let mut parts = s[1..].split(']');
+        let fg_deriv = parts
+            .next()
+            .ok_or_else(|| ScriptCreationError::DerivedKeyParsing)?;
```
Comment (darosior) [inline src/scripts.rs:96]: Oh i didn't `clippy` @danielabrozzoni will tell me to replace by `ok_or` here :sweat_smile:

File: src/scripts.rs [hunk 2]
```diff
@@ -446,6 +446,27 @@ impl DerivedDepositDescriptor {
 
         Ok(DerivedDepositDescriptor(deposit_desc!(stakeholders)))
     }
+
+    /// Get all the keys and key source used in this derived Deposit descriptor
+    pub fn keys(&self) -> Vec<DerivedPublicKey> {
```
Comment (danielabrozzoni) [inline src/scripts.rs:451]: Can't you put this method, DerivedUnvaultDescriptor::keys and DerivedCpfpDescriptor::keys directly into `impl_descriptor_newtype`, in the `derived_struct_name`? Or there are some differences between the three I'm not seeing?

File: src/scripts.rs [hunk 3]
```diff
@@ -1372,20 +1372,20 @@ mod tests {
             2018,
         )
         .expect("Valid, with xpubs");
-        assert_eq!(
```
Comment (danielabrozzoni) [inline src/scripts.rs:1111]: :joy: () == ()

Comment (darosior) [reply src/scripts.rs:1111]: https://github.com/revault/revault_tx/pull/84#issuecomment-816109948 :p

File: src/scripts.rs (hunk 2 above)
Comment (darosior) [reply src/scripts.rs]: Good call. Done in a new commit to move both `xpubs()` and `keys()`

File: src/scripts.rs (hunk 3 above)
Comment (danielabrozzoni) [reply src/scripts.rs:1111]: FAIR ENOUGH

**Review by danielabrozzoni — APPROVED** [review]
ACK f2ac20242971e23806617312b78ad3b142cb179c

**Pull request merged**